executor/join_result_generators.go

```diff
@@ -491,22 +499,26 @@ func (outputer *leftOuterJoinResultGenerator) emitToChunk(outer chunk.Row, inner
 	if len(outputer.filter) == 0 {
 		chkForJoin = chk
 	}
-	for inner := inners.Begin(); inner != inners.End(); inner = inners.Next() {
-		outputer.makeJoinRowToChunk(chkForJoin, outer, inner)
-	}
-	if len(outputer.filter) == 0 {
-		return nil
-	}
+	for inner, numToAppend := inners.Current(), outputer.maxChunkSize-chk.NumRows(); inner != inners.End() && numToAppend > 0; numToAppend = outputer.maxChunkSize - chk.NumRows() {
```

Comment:
This line is too long, how about declaring the two variable before the for loop?